File: src/main/java/uz/com/journal/app/subject/SubjectService.java
package uz.com.journal.app.subject;


import org.springframework.data.domain.Page;
import org.springframework.data.domain.Pageable;
import uz.com.journal.core.NonDeletableService;
import uz.com.platform.app.users.User;

public interface SubjectService extends NonDeletableService<Subject> {
    Page<Subject> findPage(SubjectFetch fetch, SubjectCriteria criteria, Pageable pageable, User currentUser);

    Subject findOne(String uuid, SubjectFetch fetch, User currentUser);

    Subject create(SubjectItem subjectItem, User currentUser);

    Subject update(String uuid, SubjectItem subjectItem, User currentUser);
}

File: src/main/webapp/WEB-INF/src/js/app/administration/subjects/SubjectService.js
module.exports = function ($http) {
    var _this = this;

    _this.findPage = function (page, size) {
        var params = {
            size: size || 10,
            page: page
        };
        return $http.get('/api/subjects', {params: params}).then(function (response) {
            return response.data;
        });
    };

    _this.findOne = function (uuid) {
        var params = {};
        return $http.get('/api/subjects/' + uuid, {params: params}).then(function (response) {
            return response.data;
        });
    };

    _this.create = function (subject) {
        return $http.post('/api/subjects/', subject).then(function (response) {
            return response.data;
        });
    };

    _this.update = function (subject) {
        return $http.put('/api/subjects/' + subject.uuid, subject).then(function (response) {
            return response.data;
        })
    };

    _this.delete = function (subject) {
        return $http.delete('/api/subjects/' + subject.uuid);
    }
};